morf-h2/src/main/java/org/alfasoftware/morf/jdbc/h2/H2Dialect.java

```diff
@@ -597,13 +596,10 @@ protected String getSqlForRowNumber() {
   @Override
   protected String getSqlForWindowFunction(Function function) {
     FunctionType functionType = function.getType();
-    switch (functionType) {
-      case ROW_NUMBER:
-        return "ROW_NUMBER()";
-
-      default:
-        return super.getSqlForWindowFunction(function);
+    if (functionType == FunctionType.ROW_NUMBER) {
+      return "ROW_NUMBER()";
     }
+    return super.getSqlForWindowFunction(function);
   }

```

Comment:
We want a switch statement here.
That would be consistent with other similar functions, where more cases can be expected in the future.

Reply:
I was thinking that initially, but should it be an if statement until there is actually enough options to merit a switch statement?